src/highdicom/legacy/sop.py

```diff
@@ -25,384 +31,506 @@
 
-        # Real World Value Mapping (U) - PET only
-        # TODO: http://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.7.6.16.2.html#sect_C.7.6.16.2.11  # noqa
+    def __init__(self,
+                 row_vector: ndarray,
+                 col_vector: ndarray,
+                 top_left_corner_pos: ndarray,
+                 voxel_spaceing: ndarray,
+                 dimensions: tuple) -> None:
+        """
+        Parameters
+        ----------
+        row_vector: ndarray
+            3D vector representing row of the input slice
+        col_vector: ndarray
+            3D vector representing column the input  aslice
+        top_left_corner_pos: ndarray
+            3D point representing top left coner position of the input slice
+        voxel_spaceing: ndarray
+            Three element array. 1st and 2nd copied from PexelSpacing and the
+            3rd copied from SliceThickness
+        dimensions: tuple
+            3 element tuple holding x as number of cols, y as number of rows
+            and z as 1
+        """
+        self.RowVector = row_vector
+        self.ColVector = col_vector
+        self.TopLeftCornerPosition = top_left_corner_pos
+        self.VoxelSpacing = voxel_spaceing
+        self.Dim = dimensions
 
-        perframe_items.append(perframe_item)
+    def get_normal_vector(self) -> ndarray:
+        """Returns the normal vector of the input slice
 
-        # All other attributes that are not assigned to functional groups.
-        for tag, da in ds.items():
-            if tag in assigned_attributes:
-                continue
-            elif tag in mf_attributes:
-                mf_dataset.add(da)
-            else:
-                if tag not in ignored_attributes:
-                    unassigned_dataelements[tag].append(da)
-
-    # All remaining unassigned attributes will be collected in either the
-    # UnassignedSharedConvertedAttributesSequence or the
-    # UnassignedPerFrameConvertedAttributesSequence, depending on whether
-    # values vary accross frames (original single-frame image instances).
-    unassigned_shared_ca_item = Dataset()
-    unassigned_perframe_ca_items = [
-        Dataset()
-        for _ in range(len(sf_datasets))
-    ]
-    for tag, dataelements in unassigned_dataelements.items():
-        values = [str(da.value) for da in dataelements]
-        unique_values = set(values)
-        if len(unique_values) == 1:
-            unassigned_shared_ca_item.add(dataelements[0])
+        """
+        n: ndarray = cross(self.RowVector, self.ColVector)
+        n[2] = -n[2]
+        return n
+
+    def get_distance_along_origin(self) -> float:
+        """Returns the shortest distince of the slice from the origin
+
+        """
+        n = self.get_normal_vector()
+        return float(
+            dot(self.TopLeftCornerPosition, n))
+
+    def are_parallel(
+            slice1: Any,
+            slice2: Any,
```

Comment:
Shouldn't the type annotation here be "GeometryOfSlice" instead of "Any"?

Reply:
It was already GeometryOfSlice but got an error from mypy saying that if the type is set as GeometryOfSlice why the function is doing extra type checking in "if clause" which never gets into.

Reply:
It is an unusual pattern to require a certain type parameter but then return a value correctly if the type constraint is not satisfied. This is why mypy is complaining. If you use `GeometryOfSlice` as the type annotation and then raise a `TypeError` if the inputs are the wrong type, this mypy error will go away and the behaviour of the method will be more intuitive. Type hinting using `Any` is highly misleading